addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+import * as fs from 'fs';
+import * as ts from 'typescript';
+import * as glob from 'glob';
+import * as path from 'path';
+import * as denodeify from 'denodeify';
+
+/**
+ * Declare an Interface where key is a string and value is either a string or a number.
+ * Use for storing a string literal, start position and end posittion of ImportClause node kinds.
+ */
+export interface ModuleImport {
+  specifier: string;
+  pos: number;
+  end: number;
+};
+
+export interface ModuleMap {
+  [key: string]: ModuleImport[];
+}
+
+/**
+ * Create a SourceFile as defined by Typescript Compiler API.
+ * Generate a AST structure from a source file.
+ * 
+ * @param fileName source file for which AST is to be extracted
+ */
+export function createTsSourceFile(fileName: string): Promise<ts.SourceFile> {
+  const readFile = denodeify(fs.readFile);
+  return readFile(fileName, 'utf8')
+    .then((contents: string) => {
+      return ts.createSourceFile(fileName, contents, ts.ScriptTarget.ES6, true);
+    });
+}
+
+/**
+ * @return traverses through ts.Node and returns an array of moduleSpecifiers
+ * 
+ * @param {ts.SourceFile} node: Typescript Node of whose AST is being traversed
+ * 
+ */
+export function getImportClauses(node: ts.SourceFile): ModuleImport[] {
```

Comment:
My reasoning for not having unit tests on this function is that `getDependentFiles(arg, arg)` uses this function.

But since I am also using this function for other functions that rewrites imports, I should have tests, right?